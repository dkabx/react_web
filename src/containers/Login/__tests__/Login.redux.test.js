import {
  login,
  loginSelector,
  loginInitialState,
} from '../Login.redux'
import {
  LOGIN_FAILED,
  LOGIN_REQUESTED,
  LOGIN_SUCCESSFUL,
} from '../LoginActionTypes'

describe('LOGIN REDUX', () => {
  test('Initial Login reducers state', () => {
    expect(loginInitialState).toMatchSnapshot()
  })
  test('LOGIN_REQUESTED changes state to following', () => {
    const newState = login(loginInitialState, {
      type: LOGIN_REQUESTED,
    })
    expect(newState).toMatchSnapshot()
  })
  test('LOGIN_FAILED changes state to following', () => {
    const newState = login(loginInitialState, {
      type: LOGIN_FAILED,
    })
    expect(newState).toMatchSnapshot()
  })
  test('LOGIN_SUCCESSFULL changes state to following', () => {
    const newState = login(loginInitialState, {
      type: LOGIN_SUCCESSFUL,
    })
    expect(newState).toMatchSnapshot()
  })
})
