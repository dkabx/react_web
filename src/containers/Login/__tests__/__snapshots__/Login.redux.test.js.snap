exports[`LOGIN REDUX Initial Login reducers state 1`] = `
Object {
  "isLoginRequested": false,
}
`;

exports[`LOGIN REDUX LOGIN_FAILED changes state to following 1`] = `
Object {
  "isLoginRequested": false,
}
`;

exports[`LOGIN REDUX LOGIN_REQUESTED changes state to following 1`] = `
Object {
  "isLoginRequested": true,
}
`;

exports[`LOGIN REDUX LOGIN_SUCCESSFULL changes state to following 1`] = `
Object {
  "isLoginRequested": false,
}
`;
